### PR TITLE
Brute protect content api

### PR DESCRIPTION
**Pull request by allouis**
This adds a new method to the brute middleware - for blocking brute force attacks on the content api.

We put a rate limit on the content api, and reset it per ip when a request is successful, this results in virtually no limit if you have a correct content-api key, and being blocked if you don't.

### CODE REVIEW

File: core/server/config/defaults.json [hunk 1]
```diff
@@ -57,6 +57,12 @@
             "maxWait": 3600000,
             "lifetime": 3600,
             "freeRetries":99
+        },
+        "content_api_key": {
```
Comment (allouis) [inline core/server/config/defaults.json]: These values might need tweaking. Currently you can make 100 incorrect requests in an hour before being blocked, for an hour.

File: core/server/web/shared/middlewares/api/spam-prevention.js [hunk 2]
```diff
@@ -203,10 +205,40 @@ const privateBlog = () => {
...
+
+    store = store || new BruteKnex({
+        tablename: 'brute',
+        createTable: false,
+        knex: db.knex
+    });
+
+    contentApiKeyInstance = contentApiKeyInstance || new ExpressBrute(store,
+        extend({
+            attachResetToRequest: true,
```
Comment (allouis) [inline core/server/web/shared/middlewares/api/spam-prevention.js]: This is important, so we can reset the cound if the request suceeds

File: core/server/web/shared/middlewares/brute.js [hunk 3]
```diff
@@ -73,5 +73,26 @@ module.exports = {
...
         })(req, res, next);
+    },
+
+    /*
+     * protect content api from brute force
+     */
+    contentApiKey(req, res, next) {
+        return spamPrevention.contentApiKey().getMiddleware({
+            ignoreIP: false
+        })(req, res, function (err, ...rest) {
```
Comment (allouis) [inline core/server/web/shared/middlewares/brute.js]: Here we make sure not to add the `finish` listener, unless
a) the spamPrevention middleware has finished
b) it has finished without an error

File: core/server/config/defaults.json (hunk 1 above)
Comment (allouis) [reply core/server/config/defaults.json]: Updated to be blocked starting at one hour, and moving to 24

Comment (kevinansfield) [reply core/server/config/defaults.json]: That sounds like a good starting point 👍

**Review by kevinansfield — APPROVED** [review]

**Pull request merged**